Forta-Bot-Challenge-4/COMP01/src/utils.ts

```diff
@@ -0,0 +1,47 @@
+import { LRUCache } from "lru-cache";
+import { THRESHOLD } from "./constants";
+import { BigNumber } from "@ethersproject/bignumber";
+export const amountCache: LRUCache<string, { timestamp: number; amount: any }> = new LRUCache({
+  max: 1000,
+});
+
+export const isOverThreshold = async (poolAddress: string, amount: any, timestamp: number): Promise<number> => {
+  const bigAmount = BigNumber.from(amount);
+  const bigThreshold = BigNumber.from(THRESHOLD);
+
+  // Check if there is an entry in the cache for the current poolAddress
+  if (amountCache.has(poolAddress)) {
```

Comment:
If you are only monitoring for `cUSDCv3`, then this check will always pass after the first instance, because all of the `Supply` events will be from that specific address. This would almost make the `else` block beginning at L27 redundant.

Because of that, rethink the implementation to reduce the amount of code, since it seems it could be reduced.